ote_sdk/ote_sdk/entities/model_template.py

```diff
@@ -75,22 +75,49 @@ def __str__(self) -> str:
 
 class TaskType(Enum):
     """
-    The type of algorithm within the task family.
+    The type of algorithm within the task family. Also contains relevant information
+    about the task type like label domain, if it is trainable, if it is an anomaly task
+    or if it supports global labels.
     """
 
-    NULL = auto()
-    DATASET = auto()
-    CLASSIFICATION = auto()
-    SEGMENTATION = auto()
-    DETECTION = auto()
-    ANOMALY_DETECTION = auto()
-    CROP = auto()
-    TILE = auto()
-    INSTANCE_SEGMENTATION = auto()
-    ACTIVELEARNING = auto()
-    ANOMALY_SEGMENTATION = auto()
-    ANOMALY_CLASSIFICATION = auto()
-    ROTATED_DETECTION = auto()
+    def __new__(cls, *args, **kwds):
+        obj = object.__new__(cls)
+        obj._value_ = args[0]
+        return obj
+
+    def __init__(
+        self,
+        value: int,
+        domain: Domain,
+        is_trainable: bool,
+        is_anomaly: bool,
+        is_global: bool,
+    ):
+        """
+        :param value: Unique integer for .value property of Enum (auto() does not work)
+        :param domain: Label domain of the task type
+        :param is_trainable: Boolean indicating if the task type can be trained
+        :param is_anomaly: Boolean indicating if the task is an anomaly task
+        :param is_global: Boolean indicating if the task support global labels
+        """
+        self.domain = domain
+        self.is_trainable = is_trainable
+        self.is_anomaly = is_anomaly
+        self.is_global = is_global
+
+    NULL = 1, Domain.NULL, False, False, False
+    DATASET = 2, Domain.NULL, False, False, False
+    CLASSIFICATION = 3, Domain.CLASSIFICATION, True, False, True
+    SEGMENTATION = 4, Domain.SEGMENTATION, True, False, False
+    DETECTION = 5, Domain.DETECTION, True, False, False
+    ANOMALY_DETECTION = 6, Domain.ANOMALY_DETECTION, True, True, False
+    CROP = 7, Domain.NULL, False, False, False
+    TILE = 8, Domain.NULL, False, False, False
+    INSTANCE_SEGMENTATION = 9, Domain.INSTANCE_SEGMENTATION, True, False, False
+    ACTIVELEARNING = 10, Domain.NULL, False, False, False
+    ANOMALY_SEGMENTATION = 11, Domain.ANOMALY_SEGMENTATION, True, True, False
+    ANOMALY_CLASSIFICATION = 12, Domain.ANOMALY_CLASSIFICATION, True, True, True
+    ROTATED_DETECTION = 13, Domain.ROTATED_DETECTION, True, False, False
```

Comment:
I think this is not the cleanest approach because:

1. the order of these hardcoded values depends implicitly on the order of the constructor parameters and it is not evident.
2. The lack of keys makes it difficult to read and spot potential mistakes (e.g. two bools swapped by position)

Explicit is better than implicit. My proposal is something like this:
```python
from enum import Enum
from collections import namedtuple

T = namedtuple("TaskType", "value is_trainable is_global")

class TaskType(T, Enum):
    NULL = T(value=1, is_trainable=False, is_global=False)
    CLASSIFICATION = T(value=2, is_trainable=True, is_global=True)

    def __repr__(self) -> str:
        return self.name
```

Reply:
Thanks for this suggestion. I agree that this is cleaner and will update it